exonum/src/runtime/dispatcher/schema.rs

```diff
@@ -118,6 +133,13 @@ impl<T: Access> Schema<T> {
         self.access.clone().get_proof_map(INSTANCES)
     }
 
+    /// Returns a mapping between service instance IDs and interfaces that they implement.
+    pub(crate) fn service_interfaces(
+        &self,
+    ) -> ProofMapIndex<T::Base, InstanceId, ServiceInterfaces> {
```

Comment:
I'm having doubts that using `bincode` serialization with `HashSet` and then merkelizing the values is sound.